pyanaconda/modules/storage/bootloader/systemd.py

```diff
@@ -0,0 +1,147 @@
+
+    # copy console update from grub2.py
+    def write_config_console(self, config):
+        log.info("systemd.py: write_config_console")
+        if not self.console:
+            return
+
+        console_arg = "console=%s" % self.console
+        if self.console_options:
+            console_arg += ",%s" % self.console_options
+        self.boot_args.add(console_arg)
+
+    def write_config(self):
+        log.info("systemd.py: write_config systemd start")
+        self.write_config_console(None)
+
+        # Rewrite the loader.conf
+        # For now we are just updating the timeout to actually
+        # implement the bootloader --timeout option
+        config_path = join_paths(conf.target.system_root, self.config_file)
+        log.info("systemd.py: write_config systemd loader conf : %s ", config_path)
+
+        with open(config_path, "w") as config:
+            config.write("timeout "+ str(self.timeout) + "\n")
+            config.write("#console-mode keep\n")
+
+        # update /etc/kernel/cmdline
+        # should look something like "root=UUID=45b931b7-592a-46dc-9c33-d38d5901ec29 ro resume=/dev/sda3"
+        config_path = join_paths(conf.target.system_root, "/etc/kernel/cmdline")
+        log.info("systemd.py: write_config systemd commandline : %s ", config_path)
+        with open(config_path, "w") as config:
+            args = str(self.boot_args)
+            log.info("systemd.py: systemd used boot args: %s ", args)
+
+            # pick up the UUID of the mounted rootfs,
+            root_uuid = util.execWithCapture("findmnt", [ "-sfn", "-oUUID", "/" ],
+                                             root=conf.target.system_root)
+            args += " root=UUID=" + root_uuid
+            config.write(args)
+
+        # rather than creating a mess in python lets just
+        # write the options above, and run a script which will merge the
+        # boot cmdline (after stripping inst. and BOOT_) with the anaconda
+        # settings and the kernel-install recovery/etc options.
+        rc = util.execWithRedirect(
+            "/usr/sbin/updateloaderentries",
+            [" "],
+            root=conf.target.system_root
+        )
+        if rc:
+            raise BootLoaderError(_("Failed to write boot loader configuration. "
+                                    "More information may be found in the log files stored in /tmp"))
+
+    #
+    # installation
+    #
+    def install(self, args=None):
+        log.info("systemd.py: install systemd boot install (root=%s)", conf.target.system_root)
+
+        # the --esp-path= isn't strictly required, but we want to be explicit about it.
+        rc = util.execWithRedirect("bootctl", [ "install", "--esp-path=/boot/efi",
+                                                "--efi-boot-option-description=" + productName.split("-")[0] ],
+                                   root=conf.target.system_root,
+                                   env_prune=['MALLOC_PERTURB_'])
```

Comment:
Hmm, now that is a good one. Its a copy paste job from grub2 install where it noted the dev/mapper gets garbage from the bios device names (see commit: 022b8b6b6ccc42519c3ea5aea68594c5354be2c0) if MALLOC_PERTURB is set. I think I saw this, but with one of early revisions where the BLS rootfs was garbaged up when using /dev/mapper entires. So that code has been replaced now, its probably worth retesting and removing even though its probably harmless to pull out the environment variable.